lms/djangoapps/grades/tests/test_new.py

```diff
@@ -0,0 +1,222 @@
+            display_name='Test Vertical 1'
+        )
+        problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+            question_text='The correct answer is Choice 3',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem",
+            data=problem_xml
+        )
+
+    def setUp(self):
+        super(GradeTestBase, self).setUp()
+        self.request = get_request_for_user(UserFactory())
+        self.client.login(username=self.request.user.username, password="test")
+        self.subsection_grade_factory = SubsectionGradeFactory(self.request.user)
+        self.course_structure = get_course_blocks(self.request.user, self.course.location)
+        self.course.enable_subsection_grades_saved = True
+        CourseEnrollment.enroll(self.request.user, self.course.id)
+
+
+@ddt.ddt
+class TestCourseGradeFactory(GradeTestBase):
+    """
+    Test that CourseGrades are calculated properly
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestCourseGradeFactory, cls).setUpClass()
+
+    def setUp(self):
+        super(TestCourseGradeFactory, self).setUp()
+
+    @ddt.data(
+        (True, True),
+        (True, False),
+        (False, True),
+        (False, False),
+    )
+    @ddt.unpack
+    def test_course_grade_feature_gating(self, feature_flag, course_setting):
+        # Grades are only saved if the feature flag and the advanced setting are
+        # both set to True.
+        grade_factory = CourseGradeFactory(self.request.user)
+        with patch('lms.djangoapps.grades.new.course_grade._pretend_to_save_course_grades') as mock_save_grades:
+            with patch.dict(settings.FEATURES, {'ENABLE_SUBSECTION_GRADES_SAVED': feature_flag}):
+                with patch.object(self.course, 'enable_subsection_grades_saved', new=course_setting):
+                    grade_factory.create(self.course)
+        self.assertEqual(mock_save_grades.called, feature_flag and course_setting)
+
+
+@ddt.ddt
+class SubsectionGradeFactoryTest(GradeTestBase):
+    """
+    Tests for SubsectionGradeFactory functionality.
+
+    Ensures that SubsectionGrades are created and updated properly, that
+    persistent grades are functioning as expected, and that the flag to
+    enable saving subsection grades blocks/enables that feature as expected.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(SubsectionGradeFactoryTest, cls).setUpClass()
+
+    def setUp(self):
+        """
+        Set up test course
+        """
+        super(SubsectionGradeFactoryTest, self).setUp()
+
+    def test_create(self):
+        """
+        Tests to ensure that a persistent subsection grade is created, saved, then fetched on re-request.
+        """
+        with patch(
```

Comment:
By the way, we have a `mock_passing_grade` context manager that's generally useful for this https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/tests/utils.py#L9.